src/xtal2png/core.py

```diff
@@ -970,12 +1016,13 @@ def arrays_to_structures(
 
         # build Structure-s
         S: List[Structure] = []
-        num_structures = len(atomic_numbers)
+
+        num_structures = len(atomic_symbols)
 
         for i in self.tqdm_if_verbose(range(num_structures)):
-            at = atomic_numbers[i]
+            at = atomic_symbols[i]
             fr = frac_coords[i]
-            site_ids = np.where(at > 0)
```

Comment:
As I understand, this was your criterion to identify the number of sites. However, I'm not sure if this will work so well (it will at least require some changes) as some elements will encode to zero and zero values will decode to some valid elements. 

Remedies might be to
* separately encode the number of sites 
* always ensure that sites occupied with elements are encoded > 0